spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-bus-config-sample/README.adoc

```diff
@@ -0,0 +1,119 @@
+= Spring Cloud Bus Configuration Management Demo
+
+This code sample demonstrates automated configuration management using Spring Cloud Config, Spring Cloud Bus and GCP Pub/Sub.
+The sample app consists of a configuration server and a small web application that subscribes to the server to be notified of property changes.
+
+Configuration server runs on port `8888`, and can read configuration from a local filesystem (useful for exploring the sample) or from a source control system.
+When the server discovers (or is notified via `/monitor` HTTP endpoint) that new configuration is available, it fetches the updated configuration and broadcasts `RefreshRemoteApplicationEvent` via Spring Cloud Bus.
+
+Under the hood, Spring Cloud Bus uses the Spring Cloud Stream binder for Cloud Pub/Sub to send a message to a topic named `springCloudBus`.
+To customize the topic used as a bus, use the `spring.cloud.bus.destination` property.
+If changing from the default, make sure that both server and client have the same value!
+
+The client uses the same binder with an anonymous subscription to receive notifications, and provides an HTTP endpoint `/message` for printing the dynamically updated message value.
+
+You may notice that there are two anonymous subscriptions to the `springCloudBus` topic when the sample applications are running -- one is for the client application, and the other is for the server acting as a client.
+Technically, every configuration server happens to also be a configuration client, but configuration propagation is suppressed on the server by default.
+This is why every time configuration is updated, the server will print "Received remote refresh request. Keys refreshed []", indicating that it received the bus notification, but no configuration was updated.
+
+== Setup
+
+1. Configure your GCP project ID and credentials by following link:../../docs/src/main/asciidoc/core.adoc#project-id[these instructions].
++
+Alternatively, if you have the https://cloud.google.com/sdk/[Google Cloud SDK] installed and initialized, and are logged in with https://developers.google.com/identity/protocols/application-default-credentials[application default credentials], Spring will auto-discover those parameters for you.
+
+2. Make sure that GCP Cloud Pub/Sub API is enabled, either through link:https://console.cloud.google.com/apis/api/pubsub.googleapis.com[Google Cloud Console] or with the following `gcloud` command:
+
+  gcloud services enable pubsub.googleapis.com
+
+== Demo with local filesystem
+
+=== Overview
+
+Spring Cloud Config supports retrieving configuration from a local filesystem through `native` Spring Profile that is already configured in local configuration server's link:spring-cloud-gcp-pubsub-bus-config-sample-server-local/src/main/resources/application.properties[application.properties].
```

Comment:
"`native` Spring Profile" doesn't sound right. How is it a Spring Profile?

Reply:
Spring Cloud Config uses a spring profile to enable the local filesystem functionality. So, to activate it, you set the currently active profile to "native", and then Spring Cloud Config configures (ha) itself accordingly.

Reply:
reworded to " a Spring Profile called `native`"